woodwork/tests/test_utils.py

```diff
@@ -566,3 +569,452 @@ def test_parse_logical_type_errors():
+    assert 'signup_date' in df2.columns
+
+
+def test_concat_rows_with_ww_indexes(sample_df):
+    sample_df.ww.init(index='id')
+    original_schema = sample_df.ww.schema
+    copy_df = sample_df.ww.copy()
+
+    if isinstance(sample_df, pd.DataFrame):
+        # Only pandas checks index uniqueness
+        error = 'Index column must be unique'
+        with pytest.raises(IndexError, match=error):
+            concat([sample_df, copy_df], axis=0)
+
+        # Without schema validation this does not get caught
+        combined_df = concat([sample_df, copy_df], axis=0, validate_schema=False)
+        assert combined_df.ww.index == 'id'
+
+    sample_df.ww.init(time_index='signup_date')
+    original_schema = sample_df.ww.schema
+    copy_df = sample_df.ww.copy()
+
+    combined_df = concat([sample_df, copy_df], axis=0)
+    assert len(combined_df) == 8
+
+    assert combined_df.ww.time_index == 'signup_date'
+    assert combined_df.ww.schema == original_schema
+
+
+def test_concat_table_names(sample_df):
+    df1 = sample_df[['id', 'signup_date']]
+    df1.ww.init(name=0)
+    df2 = sample_df[['full_name', 'age']]
+    df2.ww.init(name='1')
+    df3 = sample_df[['phone_number', 'is_registered']]
+    df3.ww.init(name='2')
+
+    combined_df = concat([df1, df2, df3])
+    assert combined_df.ww.name == '0_1_2'
+
+    combined_df = concat([df3, df2, df1])
+    assert combined_df.ww.name == '2_1_0'
+
+
+def test_concat_cols_different_use_standard_tags(sample_df):
+    sample_df.ww.init(use_standard_tags={'age': False, 'full_name': False, 'id': True},
+                      logical_types={'full_name': 'Categorical'})
+
+    assert sample_df.ww.semantic_tags['id'] == {'numeric'}
+    assert sample_df.ww.semantic_tags['full_name'] == set()
+    assert sample_df.ww.semantic_tags['age'] == set()
+
+    df1 = sample_df.ww[['id', 'full_name', 'email']]
+    df2 = sample_df.ww[['phone_number', 'age', 'signup_date', 'is_registered']]
+
+    combined_df = concat([df1, df2])
+
+    assert combined_df.ww.semantic_tags['id'] == {'numeric'}
+    assert combined_df.ww.semantic_tags['full_name'] == set()
+    assert combined_df.ww.semantic_tags['age'] == set()
+
+
+def test_concat_rows_different_schema(sample_df):
+    df1 = sample_df.copy()
+    df1.ww.init(logical_types={'age': 'AgeNullable'})
+    df2 = sample_df.copy()
+    df2.ww.init(logical_types={'age': 'IntegerNullable'})
+    df3 = sample_df.copy()
+    df3.ww.init(logical_types={'age': 'NaturalLanguage'})
+
+    assert df2.ww.schema != df1.ww.schema
+
+    combined_df_1 = concat([df1, df2], axis=0)
+    combined_df_1.ww.logical_types['age'] == AgeNullable
+
+    combined_df_2 = concat([df2, df1], axis=0)
+    combined_df_2.ww.logical_types['age'] == IntegerNullable
+
+    combined_df_3 = concat([df3, df2], axis=0)
+    combined_df_3.ww.logical_types['age'] == NaturalLanguage
+    assert str(combined_df_3['age'].dtype) == 'string'
+
+
+def test_concat_cols_different_schema(sample_df):
```

Comment:
To raise an error, we could store the index and time index logical types along with the name of the index and confirm that the logical types are all the same. However, this would add complexity.

To simplify, I think I'd suggest that we only allow one dataframe to have an index column set. That way, users can handle how they want to drop down to one index column instead of us handling taking just the first index column and dropping all the others.

Reply:
Maybe the result of Woodwork's concat shouldn't be any different than the result of a plain concat call. Just that we have additional requirements about no duplicate columns and that we will try to update the dtypes to keep the schema valid.

Reply:
No longer allowing multiple index columns to be specified, so we will not run into conflicting logical types that could result in this situation.